src/Uno.Extensions.Navigation.Generators/ForceBindingsUpdateGenTool_1.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Uno.Extensions.Generators;
+
+namespace Uno.Extensions.Navigation.Generators;
+
+internal class ForceBindingsUpdateGenTool_1 : ICodeGenTool
+{
+	/// <inheritdoc />
+	public string Version => "1";
+
+	private readonly ForceBindingsUpdateGenerationContext _ctx;
+	private readonly IAssemblySymbol _assembly;
+
+	public ForceBindingsUpdateGenTool_1(ForceBindingsUpdateGenerationContext ctx)
+	{
+		_ctx = ctx;
+		_assembly = ctx.Context.Compilation.Assembly;
+	}
+
+	private bool IsSupported(INamedTypeSymbol? type)
+		=> type is not null
+			&& (_ctx.IsGenerationEnabled(type)
+				?? (
+						type.Is(_ctx.Page) &&
+						type.IsPartial()
+					))
+		&& _ctx.ContainsXBind(type);
+
+	public IEnumerable<(string fileName, string code)> Generate()
+	{
+		var viewModels = from module in _assembly.Modules
+			from type in module.GetNamespaceTypes()
+			where IsSupported(type)
+			select type;
+
+		foreach (var vm in viewModels)
+		{
+			if (vm is not null)
+			{
+				yield return (vm.ToString(), Generate(vm));
```

Comment:
It *should* be the same, but clearer IMO.

```suggestion
				yield return (vm.ToDisplayString(), Generate(vm));
```

Reply:
fixed